capellambse/model/__init__.py

```diff
@@ -324,3 +328,171 @@ def search(
+            will be created. The JSON file consists of a list of
+            dictionaries (each representing a diagram in the model. The
+            dictionaries come with the keys
+
+            - name: Name of the diagram as it has been set in Capella
+            - uuid: The UUID of the diagram
+            - success: A boolean stating if a diagram has been exported
+              from Capella
+
+            The HTML file showa a numbered list of diagram names which
+            are hyperlinked to the diagram image file. Right beside a
+            diagram's name one can also see the diagram's UUID in a
+            discreet light gray and tiny font size. The HTML index also
+            provides some meta data like a date time stamp for the
+            update of diagrams.
+        force
+            For the very special case that an ambiguous value is being
+            specified for *capella_cli*. It is theoretically possible to
+            specify a *capella_cli* that matches a local Capella
+            executable and an accessible Docker image in parallel. In
+            such a case the caller can force to get the value for
+            *capella_cli* interpreted as an executable with *force* set
+            to ``"exe"`` or as a Docker image when *force* is set to
+            ``"img"`` (Docker image). If *force* is not specified and an
+            ambiguous name is specified for *capella_cli*, the local
+            executable will be used and not the accessible Docker
+            image with the same name.
+
+        Raises
+        ------
+        TypeError
+            If there has no diagram cache being specified during load of
+            the model.
+
+        Examples
+        --------
+        **Running a local installation of Capella**
+
+        All the following examples call the method
+        :meth:`update_diagram_cache` on a model
+        for which it is assumed that it has been loaded like that:
+
+        >>> model = MelodyModel(
+        ...    "/path/to/model.aird",
+        ...    diagram_cache="/path/to/diagram_cache"
+        ... )
+
+        Passing an executable/ symlink named ``capella`` that is in the
+        ``PATH`` environment variable:
+
+        >>> model.update_diagram_cache(
+        ...     "capella", "png", True
+        ... )
+
+        Passing an absolute path to a local installation of Capella that
+        contains the Capella version:
+
+        .. note::
+            Here, the algorithm will replace the placeholder
+            ``{VERSION}`` by the x.y.z. formatted version to which the
+            given model has been migrated. Therefore, for a model that
+            has been migrated to Capella 6.0, it will be tried to run
+            the executable
+            ``~/.bin/Capella_6.0.0.app/Contents/MacOS/capella``
+
+        >>> model.update_diagram_cache(
+        ...     "~/.bin/Capella_{VERSION}.app/Contents/MacOS/capella"
```

Comment:
The `~` prefix being handled specially surprises me; normally that is done by the shell and nobody else (i.e. a caller would have to do this manually, with `pathlib.Path.expanduser()` or similar mechanisms). If this is intended behavior (and not a bug), it definitely needs to be mentioned in the `capella_cli` parameter documentation.

Looking at the implementation again, this seems more like a bug to me, caused by delegating to a shell instead of executing the subprocess directly – which also means that it doesn't work on Windows. It needs to be fixed to be consistent on all OSes. I would prefer not handling `~` and letting others do that.